lib/Doctrine/DBAL/Driver/Statement.php

```diff
@@ -26,12 +61,11 @@
      *                       this will be a parameter name of the form :name. For a prepared statement
      *                       using question mark placeholders, this will be the 1-indexed position of the parameter.
      * @param mixed   $value The value to bind to the parameter.
-     * @param integer $type  Explicit data type for the parameter using the PDO::PARAM_* constants.
+     * @param integer $type  Explicit data type for the parameter using the self::PARAM_* constants.
      *
      * @return boolean TRUE on success or FALSE on failure.
      */
-    function bindValue($param, $value, $type = null);
-
+    public function bindValue($param, $value, $type = self::PARAM_STR);
```

Comment:
This is also a BC break, $type was previously nullable, now isn't.

Reply:
The parameter is not typed as `int $type`, so the callers still can pass whatever they want. Or I'm missing something?

Reply:
You're right. I was just assuming typed code in 3.x which would incorporate either `int` or `?int` typehint, and since `@param` specified `integer` (which would implicitly become nullable with `= null`), I assumed it'd become `int`. With current `mixed` it's irrelevant.